### ISSUE TITLE
Tuya-based (from Kogan) vacuum shows "unknown" instead of "cleaning"

### ISSUE THREAD

**Issue by jezzaaa** [denysdovhan/vacuum-card#395]
**Before submitting a bug**

[X] I updated to the latest version available
[X] I cleared the cache of my browser

**Describe the bug**

**Steps to reproduce**

1. Go to dashboard where the card has been setup.
2. Click on 'clean' ("play" triangle)
3. See status change from "Docked" to "unknown"
4. Click on 3-dot menu, then 'home' icon
5. See status change from "unknown" to "Returning Home" and then "Docked"

**Expected behavior**
When cleaning, "Cleaning" is expected as the status, rather than "unknown".

**Versions:**

- Release: 2.6,3
- Browser: Chrome
- HomeAssistant version: 2022.5.4

**Comment by denysdovhan** [comment]
Seems like your vacuum integration reports unsupported state to the card. Can you investigate that?

**Comment by JAKAMI99** [comment]
Same issue here, bought a Tesvor X500 and registered it over Tuya-Cloud.
Sadly, the status is also unknown.

**Comment by denysdovhan** [comment]
https://github.com/denysdovhan/vacuum-card/blob/337b272e149693f7fb2d79ca85ba80a5515bd182/src/vacuum-card.ts#L178

This is how our card determines the status of a robot.

**Comment by jezzaaa** [comment]
No updates from me for a while because I moved house and had to setup my HA installation. I added my vac the other day and it shows correctly. I'm guessing there were updates to the card that solved this problem (for me, at least).

**Comment by jezzaaa** [comment]
Hmm. I moved house again, and had to re-setup HA. The problem has returned!

Following advice on another ticket, I deleted and re-added the device using the Mode DP (27) as the ID. That has improved the situation, where the card now shows "Idle" when it's docked, and it still shows "cleaning" when it's cleaning. It would be better to show "docked" when docked, but "idle" is better than "cleaning" in this state. I note that the Tuya integration shows "docked" when the vac is docked.

The status on the card is also wrong when the vac is returning to the dock. Instead of showing "Returning to dock" (as the Tuya integration does) it shows "cleaning".
